---
title: "Week 9 DESeq analysis"
author: "Ralph Goguanco"
date: "3/4/2022"
output:
  pdf_document: default
  html_document: default
---

This week we are looking at differential expression analysis.

The data for this hands-on session comes from a published RNA-seq experiment where airway smooth muscle cells were treated with dexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects (Himes et al. 2014).

## Import/Read the data from Himes et al.

```{r}
counts <- read.csv("airway_scaledcounts.csv",
                   row.names=1)
metadata <- read.csv("airway_metadata.csv")
```

Lets have a wee peak at this data
```{r}
head(metadata)
```

Sanity check on corespondence of counts and metadata

```{r}
all(metadata$id == colnames(counts))
```
>Q1. How many genes are in this dataset? 

There are `r nrow(counts)` genes in this dataset.

>Q2. How many ‘control’ cell lines do we have?

```{r include=FALSE}
n.control <- sum(metadata$dex == "control")

```


There are `r n.control` control cell lines in this dataset.


### Extract and summarize the control samples

To find out where the control samples are we need the metadata 
```{r}
control <- metadata[metadata$dex == "control",]
control.counts <- counts[ , control$id]
control.mean <- rowMeans(control.counts)
head(control.mean)
```

>Q3. How would you make the above code in either approach more robust?

Instead of using rowSums/4, we can just use rowMeans



>Q4. Follow the same procedure for the treated samples (i.e. calculate the mean per gene across drug treated samples and assign to a labeled vector called treated.mean)

## Extract and summarize the treated (i.e. drug) samples

```{r}
treated <- metadata[metadata$dex == "treated", ]
treated.counts <- counts[, treated$id]
treated.mean <- rowMeans(treated.counts)
```

Store these results together in a new data frame called `meancounts`
Lets make a plot to explore the results a little

```{r}
meancounts <- data.frame(control.mean, treated.mean)
```

Lets make a plot to explore the results a little

```{r}
plot(meancounts[,1], meancounts[,2])
```
```{r}
library(ggplot2)

ggplot(meancounts) +
  aes(control.mean, treated.mean) +
  geom_point()
```

We will make a log-log plot to draw out this skewed data and see what is going on 

```{r}
plot(meancounts[,1], meancounts[,2], log="xy", 
    xlab="log control counts", ylab="log of treated counts")
```

We often log2 transformations when dealing with this sort of data.

```{r}
log2(80/20)

```
This log2 transformation has this nice property where if there is no change the log2 value will be zero and if it double the log2 value will be 1 and if halved it will be -1.

So lets add a log2 fold change column to our results so far

```{r}
meancounts$log2fc <- log2(meancounts$treated.mean / 
                          meancounts$control.mean)

```

```{r}
head(meancounts)

```

We need to get rid of zero count genes that we can not say anything about

```{r}
zero.vales <- (which( meancounts[,1:2]==0, arr.ind=TRUE))
to.rm <- unique(zero.vales[,1])
mycounts <- meancounts[-to.rm,]
```

```{r}
head(mycounts)
```

How many genes are remaining?

```{r}
nrow(mycounts)
```

#Use fold change to see up and down regulated genes.

Acommon threshold used for calling something differentially expressed is a log2(FoldChange) of greater than 2 or less than -2. Let's filter the dataset both ways to see how many genes are up or down-regulated

```{r}
sum(mycounts$log2fc > 2)

```
and d own regulated
```{r}
sum(mycounts$log2fc < -2)
```

# DESeq2 analysis

Let's do this the right way. DESeq2 is an R package specifically for analyzing count-based NGS data like RNA-seq.

```{r}
# load up  DESeq2
library(DESeq2)

dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex)
```

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

We can get some basic summary tallies using the `summary()` function

```{r}
summary(res, alpha=0.05)
```

#Volcano plot

Make a summary plot of our results.

```{r}
plot(res$log2FoldChange, -log(res$padj))
```

Finish for today by saving our results 

```{r}
write.csv(res, file="DESeq2_results.csv")
```

























